src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,54 @@
-import React from 'react';
+import { FC, useEffect, useState } from 'react';
+import classNames from 'classnames';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+interface TodoModalProps {
+  handleCloseModal: () => void;
+  selectedTodo: Todo;
+}
+
+export const TodoModal: FC<TodoModalProps> = ({
+  handleCloseModal,
+  selectedTodo,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+
+  const {
+    title,
+    completed,
+    userId,
+    id,
+  } = selectedTodo;
+
+  const fetchUsers = async () => {
+    setIsLoading(true);
+
+    try {
+      setUser(await getUser(userId));
+    } catch {
+      setError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchUsers();
+  }, [selectedTodo]);
```

Comment:
why do you need selectedTodo inside of the dependencies array? if you close modal, it will unmount, and then mount when you select other todo, won't it work just fine without it?